src/core/json/path.cc

```diff
@@ -439,7 +66,7 @@ void EvaluatePath(const Path& path, const JsonType& json, PathCallback callback)
   }
 
   if (path.front().type() != SegmentType::FUNCTION) {
-    Dfs().Traverse(path, json, std::move(callback));
+    detail::Dfs().Traverse(path, json, std::move(callback));
```

Comment:
As I write earlier consider to make static function Traverse and mutate that will return Dfs object

Reply:
static Dfs Traverse(...) {
Dfs res;
res,Traverse(...)
return res;
}

Reply:
but that would leak Dfs in the interface. why is it better?

Reply:
Ah, I misunderstood you, will do it in the next PR

Reply:
I suggest to add this method into Dfs class. because you need to call Traverse of mutate on the temporary objects, or you often do `Dfs dfs; dfs.Mutate()`